java/src/org/openqa/selenium/manager/SeleniumManager.java

```diff
@@ -0,0 +1,128 @@
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package org.openqa.selenium.manager;
+
+import com.google.common.io.CharStreams;
+import org.openqa.selenium.Platform;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.logging.Logger;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+import static org.openqa.selenium.Platform.MAC;
+import static org.openqa.selenium.Platform.WINDOWS;
+
+public class SeleniumManager {
+
+    private static final Logger LOG = Logger.getLogger(SeleniumManager.class.getName());
+
+    private static final String SELENIUM_MANAGER = "selenium-manager";
+    private static final String EXE = ".exe";
+    private static final String INFO = "INFO\t";
+
+    private static SeleniumManager manager;
+
+    private File binary;
+
+    private SeleniumManager() {
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            if (binary != null && binary.exists()) {
+                try {
+                    Files.delete(binary.toPath());
+                } catch (IOException e) {
+                    LOG.warning(String.format("%s deleting temporal file: %s",
+                            e.getClass().getSimpleName(), e.getMessage()));
+                }
+            }
+        }));
+    }
+
+    public static SeleniumManager getInstance() {
```

Comment:
Is there a reason to use a static factory method? We don't tend to do this in other parts of Selenium code, and (for now) I'm thinking of this class as "private api" / not something users will work with directly...

Reply:
Actually, in Python & Ruby I'm implementing `getDriverPath()` as a static method directly, so they don't instantiate the object at all. Would that be limiting if we did it in Java?

Reply:
This is an implementation of the singleton design pattern, which fits nicely in this case (see the following comment).

Reply:
oh, this makes much more sense than what I thought this was doing.